### ISSUE TITLE
Can't have a field called `catch`

### ISSUE THREAD

**Issue by Seelengrab** [JuliaLang/julia#53666]
MWE:

```julia
julia> struct Foo
           catch::Bool
       end
ERROR: ParseError:
# Error @ REPL[3]:2:5
struct Foo
    catch::Bool
#   └ ── Expected `end`
Stacktrace:
 [1] top-level scope
   @ none:1
```

`catch` is only a keyword in conjunction with a preceding `try`, so IMO this should work.

**Comment by Seelengrab** [comment]
The same is true for `catch` is a keyword to a function too.

**Comment by oscardssmith** [comment]
I think the problem is that if `catch` were a valid function name,

```
catch() = 1
try
catch
catch
end
```
would be ambiguous

**Comment by Seelengrab** [comment]
That's an ambiguity the parser can easily decide one way or another though, since it doesn't parse right now and nothing can break because of that.

**Comment by andrewjradcliffe** [comment]
I think that it is best to avoid the binding of variables to keywords, particularly when the keywords are related to control flow -- it drastically decreases code readability, even with syntax highlighting which disambiguates the use cases.

**Comment by jariji** [comment]
> I think that it is best to avoid the binding of variables to keywords

Yes, though if possible that should probably be the user's decision.

**Comment by Seelengrab** [comment]
> I think that it is best to avoid the binding of variables to keywords,

There is precedent for allowing this though: https://github.com/JuliaLang/julia/pull/53685 and https://github.com/JuliaLang/julia/pull/53341

**Comment by barucden** [comment]
> > I think that it is best to avoid the binding of variables to keywords,
> 
> There is precedent for allowing this though: #53685 and #53341 (`throw` is a bit special, it's not really a function even though it claims to be..)

`throw` is not a keyword though
```julia-repl
julia> throw = 42
42
```

> My use case is similiar to those two - I want to store whether an operation done on that struct should `catch` or propagate an error.

Isn't it better to name that field `should_catch` anyway?

**Comment by Seelengrab** [comment]
I prefer `catch`, as if the object is asking "Do I `catch` errors?". `should_catch` leaves the possibility open that even if it should, it doesn't necessarily always do that.

**Comment by BioTurboNick** [comment]
Should this be closed? No keywords are allowed to be used as names, and that seems unlikely to ever change.

**Comment by BioTurboNick** [comment]
Also, you can do this:
```
struct Foo
    var"catch"::Bool
end
Foo(true).catch
```

(Literally just thought to try this and it... just worked.)

EDIT: You can also make a function called `catch` this way, but it can't be called without wrapping in `var""`

**Comment by Seelengrab** [comment]
Hmm it's not great, but I guess it's a compromise - thanks!
